src/Interpreters/PredicateExpressionsOptimizer.cpp

```diff
@@ -49,13 +49,6 @@ static ASTs splitConjunctionPredicate(const std::initializer_list<const ASTPtr>
 {
     std::vector<ASTPtr> res;
 
-    auto remove_expression_at_index = [&res] (const size_t index)
```

Comment:
@alexey-milovidov 
I think this is an unnecessary change, and the modify predicates order should be reasonable.
We keep it so that we can continue to find same bugs.

Reply:
I don't have any strong arguments, but executing different queries on the initiator and remote shards (even if only the order differs) sounds tricky

But catching some possible issues maybe a good argument to ignore trickery

Reply:
@alexey-milovidov friendly ping